teuthology/test/test_matrix.py

```diff
@@ -0,0 +1,70 @@
+from .. import matrix
+
+def do_test(res):
```

Comment:
I dislike this method.  It is very obfuscated and difficult to understand what is actually being asserted here.  For example, if test_product3 failed below I'd be very confused as to why it failed or what should have been expected.